src/components/controls/common/ClearButton/ClearButton.scss

```diff
@@ -3,8 +3,14 @@
 $block: '.#{variables.$ns-new}clear-button';
 
 #{$block} {
-    --g-button-text-color: var(--g-color-text-hint);
-    --g-button-text-color-hover: var(--g-color-text-primary);
+    --g-button-text-color: var(
+        --g-text-input-placeholder-color,
+        var(--g-text-area-placeholder-color, var(--g-color-text-hint))
+    );
+    --g-button-text-color-hover: var(
+        --g-text-input-text-color,
+        var(--g-text-area-text-color, var(--g-color-text-primary))
+    );
```

Comment:
Revert this please, button colors should not be affected by text colors of input